src/legacy/core_plugins/kibana/public/management/sections/settings/components/component_registry.ts

```diff
@@ -17,7 +17,10 @@
  * under the License.
  */
 
-const registry = {};
+import { FunctionComponent } from 'react';
+
+type Id = string;
+const registry: Record<Id, FunctionComponent> = {};
```

Comment:
Wouldn't this registry be able to accept any type of a React component and not just `FunctionComponent` ?